include/cst_tokenstream.h

```diff
@@ -77,7 +77,11 @@ typedef struct cst_tokenstream_struct {
     const cst_string *p_prepunctuationsymbols;
     const cst_string *p_postpunctuationsymbols;
 
+
     cst_string charclass[256];
+    cst_string *charclass2[256];
+    cst_string **charclass3[256];
+    cst_string ***charclass4[256];
 
```

Comment:
I think technically the arrays for charclass 2-4 doesn't need to be more than 64 since utf8 only uses 6 bits of the latter characters (Could probably be shortened even more since the first character usage is even smaller for multi-byte characters). This would probably require an extra operation when addressing the charclass `c[x] & 0x3f` when filtering out the class. I guess it's a performance versus memory decision.

Another thing to consider is that I might have misunderstood something completely and the entire span is needed.

Related question, how many of these would actually be used for marking the special character classes?